app/src/main/java/ch/blinkenlights/android/vanilla/FileUtils.java

```diff
@@ -34,6 +34,124 @@
  * Provides some static File-related utility functions.
  */
 public class FileUtils {
+	/**
+	 * The string (but also valid path!) to use to indicate the current directory
+	 */
+	public static final String NAME_CURRENT_FOLDER = ".";
+	/**
+	 * The string (but also valid path!) to use to indicate the parent directory
+	 */
+	public static final String NAME_PARENT_FOLDER = "..";
+
+	private FileUtils() {
+		// Private constructor to hide implicit one.
+	}
+
+	/**
+	 * Converts common path separator characters ('/' or '\\') to the value of File.separatorChar.
+	 *
+	 * @param path The path to convert directory separators within.
+	 * @return A path with all valid separators normalized to the system's.
+	 */
+	public static String normalizeDirectorySeparators(String path) {
+		final StringBuilder sb = new StringBuilder(path);
+
+		for (int i = 0; i < sb.length(); i++) {
+			final char originalChar = sb.charAt(i);
+			if (originalChar == '/' || originalChar == '\\') {
+				sb.setCharAt(i, File.separatorChar);
+			}
+		}
+		return sb.toString();
+	}
+
+	/**
+	 * Utility function to prepend n-parent directory traversals to a path.
+	 *
+	 * @param path A relative path to traverse upwards from within.
+	 * @param traversalCount The number of directories to navigate upwards.
+	 * @return The result of creating the relative path.
+	 */
+	private static String traversePathUpwards(String path, int traversalCount) {
+		if (traversalCount < 0)
+			throw new IllegalArgumentException("Cannot descend into the directory structure.");
+
+		if (traversalCount == 0)
+			return path;
+
+		StringBuilder sb = new StringBuilder((FileUtils.NAME_PARENT_FOLDER.length() + 1) * traversalCount +
+			path.length());
+
+		for (int i = 0; i < traversalCount; i++) {
+			sb.append(FileUtils.NAME_PARENT_FOLDER);
+			sb.append(File.separatorChar);
+		}
+		sb.append(path);
+		return sb.toString();
+	}
+
+	/**
+	 * Attempts to generate a relative path from a source File to a destination.
+	 * Will attempt to perform directory traversal to find a common path.
+	 * Upon failure to relativize, returns the original destination.
+	 *
+	 * @param source The File the path is relative to e.g. /a/b/c/
```

Comment:
'source' is supposed to point to a folder, right?
Maybe call it 'base' instead of source?